test/apps/process/wait4.c

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#include "../network/test.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <signal.h>
+#include <linux/wait.h>
+#include <pthread.h>
+#include <sys/mman.h>
+
+static pid_t pid;
+static int status;
+
+FN_SETUP(fork_child)
+{
+	pid = CHECK(fork());
+
+	if (pid == 0) {
+		// Child entering an infinite loop until killed by parent.
+		while (1) {
+			usleep(100);
+		}
+
+		exit(EXIT_SUCCESS);
+	}
+
+	// Parent process
+	sleep(1); // Ensure the child process is running
+}
+END_SETUP()
+
+FN_TEST(stop_child)
+{
+	// Stop the child process
+	TEST_SUCC(kill(pid, SIGSTOP));
+	TEST_RES(wait4(pid, &status, WSTOPPED, NULL),
+		 _ret == pid && WIFSTOPPED(status) &&
+			 WSTOPSIG(status) == SIGSTOP);
+	status = 0;
+	TEST_RES(wait4(pid, &status, WSTOPPED | WNOHANG, NULL),
+		 _ret == 0 && !WIFSTOPPED(status));
+}
+END_TEST()
+
+FN_TEST(continue_child)
+{
+	TEST_SUCC(kill(pid, SIGCONT));
+	status = 0;
+	TEST_RES(wait4(pid, &status, WCONTINUED, NULL),
+		 _ret == pid && WIFCONTINUED(status));
+	status = 0;
+	TEST_RES(wait4(pid, &status, WCONTINUED | WNOHANG, NULL),
+		 _ret == 0 && !WIFCONTINUED(status));
```

Comment:
It seems that you sometimes write:
```c
		 _ret == 0 && !WIFCONTINUED(status));
		 _ret == 0 && !WIFSTOPPED(status));
```
sometimes write:
```c
		 _ret == 0 && status == 0);
```

I think we can unify them to one style instead. Maybe just checking against 0.